vm/src/stdlib/time_module.rs

```diff
@@ -36,22 +36,24 @@ fn duration_to_f64(d: Duration) -> f64 {
     (d.as_secs() as f64) + (f64::from(d.subsec_nanos()) / 1e9)
 }
 
-fn time_time(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args);
-    let x = match SystemTime::now().duration_since(UNIX_EPOCH) {
+fn get_time() -> f64 {
+    match SystemTime::now().duration_since(UNIX_EPOCH) {
         Ok(v) => duration_to_f64(v),
-        Err(err) => panic!("Error: {:?}", err),
-    };
-    let value = vm.ctx.new_float(x);
-    Ok(value)
+        Err(err) => panic!("Time error: {:?}", err),
+    }
+}
+
+fn time_time(vm: &VirtualMachine) -> f64 {
+    let time_func = vm.settings.syscall_functions.time.unwrap_or(get_time);
```

Comment:
Instead of adding a `syscall_functions` member, create two `time_time` functions and select them with the `#cfg` structure. If you want to reduce duplicated code, you can create a helper function which you implement for each platform.

It might even make sense to create specific platform subfolders in the `stdlib` folder with the specifics for each platform.